### PR TITLE
Default mods (autoinit_ruby and language_ruby)

### LINKED ISSUE #5: Install language-ruby and autoint-ruby mods when creating a new project
Running
```console
gotta new project new-project
```
should automatically install the `language-ruby` and `autoinit-ruby` mods

### PR DESCRIPTION
**Pull request by tylerklose**
## What does this PR do?
Installs `autoinit_ruby` and `language_ruby` as default mods when you create a new project. Also cleans up how we interact with the file system. By going through the Ruby classes, we reduce our chances of running into cross-platform issues

## Do you have any concerns about this body of work? Anything you'd like to highlight or open for discussion? Anything to be aware of before merging / deploying?
Nope

## Where should the review start?
`lib/gotta/mod/repository.rb`
`lib/gotta/project/builder.rb`
## Recommendation/Steps for how to test this
`gotta new project coming-soon`
`ls coming-soon/mods`

You should see both `autoinit_ruby` and `language_ruby` there

## Link to the related issue(s)
Closes #5

### CODE REVIEW

File: lib/gotta/mod/cli/install.rb [hunk 1]
```diff
@@ -39,7 +39,7 @@ def mod_exists?
         end
 
         def github
-          option '--github'
+          option '--github', :github
```
Comment (parruda) [inline lib/gotta/mod/cli/install.rb]: @tylerklose are you intentionally creating a command option `:github`? Like `gotta install mod :github URL`?

File: lib/gotta/mod/repository.rb [hunk 2]
```diff
@@ -35,41 +37,54 @@ def url
...
+        teardown_local_repo(project_config, temp_directory)
       end
 
       private
 
       def retrieve_mod(project_config)
-        `cd #{repo_name} && git pull --depth=1 origin master`
-        if !Dir.exist?("#{repo_name}/#{path}") || Dir.empty?("#{repo_name}/#{path}")
-          puts "#{repo_name}/#{path} does not exist. Please make sure you provided the correct file structure."
+        Dir.chdir("#{Dir.pwd}/#{repo_name}")
```
Comment (parruda) [inline lib/gotta/mod/repository.rb]: The downside of doing a straight up `Dir.chdir` is that the whole ruby process will switch to that new working directory. That means any file operations after that point that use relative paths will have to take that in consideration.

`Dir.chdir` takes a block and wraps it in 2 calls to CD into the directory and then CD back into the previous dir.
 
```ruby
Dir.chdir('/usr/local') do
  # do stuff in /usr/local
end
# no need for Dir.chdir('..') or similar
```

File: lib/gotta/mod/repository.rb [hunk 3]
```diff
@@ -35,41 +37,54 @@ def url
...
+        destination = [project_config.dir, project_config.mods.folder, mod_name].join('/')
+
+        FileUtils.remove_dir(destination) if Dir.exists?(destination)
+        FileUtils.cp_r(path, destination)
       end
 
-      def setup_local_repo
-        `git init #{repo_name}`
+      def setup_local_repo(temp_directory)
+        Dir.chdir(temp_directory)
```
Comment (parruda) [inline lib/gotta/mod/repository.rb]: Same comment here about the `Dir.chdir`

File: lib/gotta/mod/cli/uninstall.rb [hunk 4]
```diff
@@ -34,7 +34,7 @@ def uninstall(args)
 
           project = Gotta::Project::Config.load
 
-          `rm -rf #{project.mods.folder}/#{mod_name}`
+          FileUtils.remove_dir("#{project.mods.folder}/#{mod_name}")
```
[the remark this answers is not in the document]
Comment (tylerklose) [reply lib/gotta/mod/cli/uninstall.rb]: That's a great suggestion, I will write something to the console!

File: lib/gotta/mod/cli/install.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (tylerklose) [reply lib/gotta/mod/cli/install.rb]: The intention is to have `gotta install mod :mod_name --github :github`

File: lib/gotta/mod/repository.rb (hunk 2 above)
Comment (tylerklose) [reply lib/gotta/mod/repository.rb]: That's way cleaner, I will do that! Thanks

File: lib/gotta/mod/cli/install.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (parruda) [reply lib/gotta/mod/cli/install.rb]: You don't need to specify the `:github` there like when you create a command.
For options, the value is already assumed to be what is passed to the option on the command line.
The method `option` already returns the value for that option.

**Comment by tylerklose** [comment]
@parruda I believe I resolved your concerns. Ready for another pass

**Pull request merged**